src/Spd.Resource.Repository/Licence/LicenceRepository.cs

```diff
@@ -17,15 +18,23 @@ internal class LicenceRepository : ILicenceRepository
 
     public async Task<LicenceResp> GetAsync(Guid licenceId, CancellationToken ct)
     {
-        spd_licence? licence = await _context.spd_licences
-            .Expand(i => i.spd_LicenceHolder_contact)
-            .Expand(i => i.spd_LicenceHolder_account)
-            .Expand(i => i.spd_CaseId)
-            .Where(l => l.spd_licenceid == licenceId)
-            .FirstOrDefaultAsync(ct);
-
-        if (licence == null)
-            throw new ArgumentException($"cannot find the licence with licenceId : {licenceId}");
+        spd_licence? licence;
+        try
+        {
+            licence = await _context.spd_licences
+                .Expand(i => i.spd_LicenceHolder_contact)
+                .Expand(i => i.spd_LicenceHolder_account)
+                .Expand(i => i.spd_CaseId)
+                .Where(l => l.spd_licenceid == licenceId)
+                .FirstOrDefaultAsync(ct);
+        }
+        catch (DataServiceQueryException ex)
+        {
+            if (ex.Response.StatusCode == 404)
+                throw new ArgumentException($"cannot find the licence with licence Id : {licenceId}");

```

Comment:
No, Ruben, what we want is when there is no record found, we return null.

Reply:
Fixed